### DISCUSSION TITLE
email says we need to set up 2FA, but setting up 2FA does not work ... what to do?

### DISCUSSION THREAD

**Discussion by Reasoning-Technology** [community/community#69428] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

A good friend has said the quiet part out loud, that companies purposely create artificial time wasters to taunt clients.  This experience shows him to not be a cynic.

So the email arrives, that my github must have 2FA.   I don't mind, I've tried to set it up before, but it did not work, and it still doesn't.

1. email says my cryptographic hard password is not good enough I need 2FA.
2. I login to setup up my hardware key.  They say that is not good enough, that I must setup SMS first (3FA?)
3. I go to setup SMS and they say they sent me an authentication code at the number.  Problem is that no authentication code arrives.
4. help instruction say that I must configure the phone number into the account first or the authentication code will not arrive.  How clever of github not to mention this in the process of asking for a number to send the 2FA code to.  Is this a circular probem?
5. so I go to [picture]->settings->account.   Options are: change username, link patreon account, export account data, successor,  and delete account.   No where is there an option to enter a phone number.  Nor is there such an option on the profile or other settings.
6. ask for support, look there is a category for 2FA settings (they must get a lot of these),  it takes me through a question tree that basically asks if 2FA is working  - and it is not or I wouldn't be on that page, what is the point? At the end of the tree they ask me to leave feedback for github.  How quaint, and unhelpful.  Someone over at MS must have been seen hysterically laughing in the corridors after inventing that trap. 
7. Hit the new ticket button, end up back at the tree, same result.  Insist on a contact, and it takes me to the discussion forum.

So what happens here?  Can you make the button for setting up the hardware key appear?   That would be great.

What a waste of time ...   why is the setup hardware key not on the 2FA menu in the first place?   What company would send a mass email out telling all the users that they have to setup 2FA, or their accounts will not be accessible, when they don't have the hardware key option on their 2FA menu?  I have to agree with my cynical friend, an oversight like this is too great to be an accident.  And what is this with not sending out the authentication code for SMS?  Same.

**Comment by ghost** [comment]
Github is harvesting user info probably to sell to third parties, Micro$oft needs that return on investment money.

Migrate over to Gitlab or another vendor, Github has lost their minds.

**Comment by DNin01** [comment]
It seems like SMS isn't working for you. In that case, you might try [configuring 2FA using a TOTP mobile app](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication#configuring-two-factor-authentication-using-a-totp-mobile-app). It's currently the only other possible primary option.

You might also get some value out of GitHub's [2FA FAQ post](https://github.com/orgs/community/discussions/68240).
